src/EPR.Payment.Service.UnitTests/Services/RegistrationFees/ProducerFeesCalculatorServiceTests.cs

```diff
@@ -0,0 +1,401 @@
+﻿using AutoFixture.MSTest;
+using EPR.Payment.Service.Common.Dtos.Request.RegistrationFees;
+using EPR.Payment.Service.Common.Dtos.Response.RegistrationFees;
+using EPR.Payment.Service.Common.UnitTests.TestHelpers;
+using EPR.Payment.Service.Services.Interfaces.RegistrationFees;
+using EPR.Payment.Service.Services.RegistrationFees;
+using EPR.Payment.Service.Strategies.Interfaces.RegistrationFees;
+using EPR.Payment.Service.Utilities.RegistrationFees.Interfaces;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using FluentValidation;
+using FluentValidation.Results;
+using Moq;
+
+namespace EPR.Payment.Service.UnitTests.Services.RegistrationFees
+{
+    [TestClass]
+    public class ProducerFeesCalculatorServiceTests
+    {
+        private Mock<IFeeCalculationStrategy<ProducerRegistrationFeesRequestDto>> _baseFeeCalculationStrategyMock = null!;
+        private Mock<IFeeCalculationStrategy<ProducerRegistrationFeesRequestDto>> _subsidiariesFeeCalculationStrategyMock = null!;
+        private Mock<IValidator<ProducerRegistrationFeesRequestDto>> _validatorMock = null!;
+        private Mock<IFeeBreakdownGenerator<ProducerRegistrationFeesRequestDto, RegistrationFeesResponseDto>> _feeBreakdownGeneratorMock = null!;
+        private IProducerFeesCalculatorService _calculatorService = null!;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            _baseFeeCalculationStrategyMock = new Mock<IFeeCalculationStrategy<ProducerRegistrationFeesRequestDto>>();
+            _subsidiariesFeeCalculationStrategyMock = new Mock<IFeeCalculationStrategy<ProducerRegistrationFeesRequestDto>>();
+            _validatorMock = new Mock<IValidator<ProducerRegistrationFeesRequestDto>>();
+            _feeBreakdownGeneratorMock = new Mock<IFeeBreakdownGenerator<ProducerRegistrationFeesRequestDto, RegistrationFeesResponseDto>>();
+
+            _calculatorService = new ProducerFeesCalculatorService(
+                _baseFeeCalculationStrategyMock.Object,
+                _subsidiariesFeeCalculationStrategyMock.Object,
+                _validatorMock.Object,
+                _feeBreakdownGeneratorMock.Object
+            );
+        }
+
+        [TestMethod]
+        public void Constructor_ShouldThrowArgumentNullException_WhenBaseFeeCalculationStrategyIsNull()
+        {
```

Comment:
naming convention we are following is methodundertest_scenario_expectedresult. please change at all the places where it is not followed.